### ISSUE TITLE
Option for Rake task to publish to repositories other than RubyGems.org

### ISSUE THREAD

**Issue by dnrce** [bundler/bundler-features#81]
The `release` Rake task publishes the built gem by calling the shell command `gem push`, which is specific to RubyGems.org. There is currently no way to override this without some sketchy monkey-patching.

Feature request: expose the `release` (sub)tasks such that they can be overridden, or provide another means of configuring the command used in the publish step.

Some third-party repository types and their "publish" commands:
* Gemfury: `fury push`
* Gem in a Box: `gem inabox`
* Nexus: `gem nexus`

**Comment by indirect** [comment]
We'd like to do this! Exposed and overwritable rake tasks would be my preference. Want to send a pull request? :)

**Comment by bronzdoc** [comment]
I would like to work on this, but need more background about it

**Comment by segiddins** [comment]
@bronzdoc the rake tasks just need to be customizable in general

**Comment by indirect** [comment]
@bronzdoc it should be possible to do something like this in your Rakefile:

```ruby
# Rakefile
require 'bundler/gem_tasks'
Rake::Task['release'].clear
task :release do
  sh("something non default")
end
```

That's all... but you can't do that now :sweat_smile:

**Comment by bronzdoc** [comment]
ok, so it's just: allow people overwrite bundler gem tasks, it makes sense now, i'll give it a shot.

**Comment by bronzdoc** [comment]
Just gave it a shot at this and there is nothing to do... you can already overwrite bundler's gems tasks
like the example provided by @indirect
```ruby
# Rakefile
require 'bundler/gem_tasks'
Rake::Task['release'].clear
task :release do
  sh("echo doc")
end
```

**Comment by segiddins** [comment]
IMHO the tasks should be configurable without completely overwriting them

**Comment by indirect** [comment]
@segiddins the only requested change so far is to completely replace the command that is run by the task... I don't know how to make that configurable without also completely replacing the API :/

**Comment by dnrce** [comment]
Yep, this definitely fulfills my wishes. It seems like it was fixed in 1.8 via #3073, a couple weeks before this ticket was filed -- not sure why I didn't realize it was there. Thanks, everyone!

**Comment by dnrce** [comment]
One thing that I don't think is possible to do via this method: allow the overridden task to use the `built_gem_path`, which is pretty important. Now I'm wondering whether my original request was more along the lines of making the methods in `GemHelper` part of the public API so they could be overridden and still receive that argument.
